src/borg/helpers/shellpattern.py

```diff
@@ -60,7 +61,86 @@ def translate(pat, match_end=r"\Z"):
                 elif stuff[0] == "^":
                     stuff = "\\" + stuff
                 res += "[%s]" % stuff
+        elif c == "(" or c == "|" or c == ")":
+            if i > 0 and pat[i - 1] != "\\":
+                res += c
         else:
             res += re.escape(c)
 
     return "(?ms)" + res + match_end
+
+
+def _parse_braces(pat):
+    """Returns the index values of paired braces in `pat` as <key, value> pairs, else None.
```

Comment:
`... as <key, value> pairs ...`

That sounds like a list of tuples, but this returns a dict mapping curly brace opening index to corresponding curly brace closing index. But please first see the code feedback below...